File: FlappyBird/Bird.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FlappyBird
{
    class Bird
    {
        // Properties 
        int size;
        double posX;
        double posY;

        // Mechanics behind bird flight
        double vel = 0;
        double gravity = 2;
        double upthrust = -25;

        // Constructor:
        public Bird(int s, int x, int y)
        {
            this.size = s;
            this.posX = x;
            this.posY = y;
        }

        // Setter Methods:
        public void SetS(int s)
        {
            this.size = s;
        }

        public void SetX(int x)
        {
            this.posX = x;
        }

        public void SetY(int y)
        {
            this.posY = y;
        }

        // Getter Methods:
        public int GetSize()
        {
            return this.size;
        }

        public double GetX()
        {
            return this.posX;
        }

        public double GetY()
        {
            return this.posY;
        }

        // Custom Methods:
        public void Fly(int height)
        {
            this.vel += this.gravity;
            this.vel *= 0.9;
            this.posY += this.vel;

            if (this.posY > height)
            {
                this.posY = height;
                this.vel = 0;
            }
        }

        public void Jump()
        {
            this.vel += this.upthrust;
        }
    }
}

File: JavaScript Version/bird.js
function bird(x, y) {
  this.posX = x;
  this.posY = y;
  this.size = 20;
  this.vel = 0;
  this.gravity = 1;
  this.upthrust = 20;

  this.updatePos = function() {
    this.vel += this.gravity;
    this.vel *= 0.9;
    this.posY += this.vel;

    if (this.posY + this.size / 2 > height) {
      this.posY = height - this.size / 2;
      this.vel = 0;
    }
  }

  this.jump = function() {
    this.vel += -this.upthrust;
  }

  this.show = function() {
    ellipse(this.posX, this.posY, this.size, this.size);
  }
}
